plugins/factory-plugin/src/workspace-projects-manager.ts

```diff
@@ -61,9 +54,16 @@ abstract class WorkspaceProjectsManager {
         }
 
         theia.window.showInformationMessage('Che Workspace: Starting cloning projects.');
+        const workspaceFolders: theia.Uri[] = [];
         await Promise.all(
-            cloneCommandList.map(cloneCommand => cloneCommand.execute())
+            cloneCommandList.map(cloneCommand => {
+                if (!cloneCommand.isInTheiaWorkspace()) {
+                    workspaceFolders.push(theia.Uri.file(cloneCommand.folder));
+                }
+                return cloneCommand.clone();
+            })
         );
+        await theia.commands.executeCommand('che.workspace.addFolder', workspaceFolders);
```

Comment:
same comments i had for factory-initializer

> I am not in favour of having logic implementation here. It should go to the command implementation.

You see here you are duplicating code.
Also, workspace root folders may be handle differently in the future see: https://github.com/eclipse/che/issues/15347